### PR TITLE
Increase maximum size for GrowableArray<T>

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

**Comment by brianrob** [comment]
@sharwell, this seems reasonable.  One question - in addition to increasing the maximum size, this also is a bit more aggressive at increasing the size of the array.  Is this for very large traces or heaps where you're seeing lots of calls to realloc?

**Comment by sharwell** [comment]
> this also is a bit more aggressive at increasing the size of the array

This was not intentional. My only goal was to latch the size at `int.MaxValue` where the previous code would overflow somewhere around `int.MaxValue/3`.

File: src/FastSerialization/GrowableArray.cs [hunk 1]
```diff
@@ -399,15 +399,20 @@ private void Realloc(int minSize)
             }
             else
             {
-                int expandSize = array.Length * 3 / 2 + 8;
-                if (minSize < expandSize)
+                long expandSize = (long)array.Length * 3 / 2 + 8;
+                if (minSize > expandSize)
```
Comment (brianrob) [inline src/FastSerialization/GrowableArray.cs:403]: Would you mind reverting `>` back to `<`?  Otherwise, this change looks good to me.

Comment (sharwell) [reply src/FastSerialization/GrowableArray.cs]: Update to closer match the original

**Review by brianrob — APPROVED** [review]

**Pull request merged**